kv/txn_coord_sender.go

```diff
@@ -220,9 +244,62 @@ func NewTxnCoordSender(wrapped client.Sender, clock *hlc.Clock, linearizable boo
 		linearizable:      linearizable,
 		stopper:           stopper,
 	}
+
+	tc.stopper.RunWorker(tc.startStats)
 	return tc
 }
 
+// startStats blocks and periodically logs transaction statistics (throughput,
+// success rates, durations, ...).
+// TODO(mrtracy): Add this to TimeSeries.
+func (tc *TxnCoordSender) startStats() {
+	res := time.Millisecond // for duration logging resolution
+	lastNow := tc.clock.PhysicalNow()
+	for {
+		select {
+		case <-time.After(statusLogInterval):
+			if !log.V(1) {
+				continue
+			}
+
+			tc.Lock()
+			curStats := tc.txnStats
+			tc.txnStats = txnCoordStats{}
+			tc.Unlock()
+
+			now := tc.clock.PhysicalNow()
+
+			// Tests have weird clocks.
+			if now-lastNow <= 0 {
+				continue
+			}
+
+			num := len(curStats.durations)
+			dMax := time.Duration(stats.Max(curStats.durations))
+			dMean := time.Duration(stats.Mean(curStats.durations))
+			dDev := time.Duration(stats.StdDevP(curStats.durations))
+			rMax := stats.Max(curStats.restarts)
+			rMean := stats.Mean(curStats.restarts)
+			rDev := stats.StdDevP(curStats.restarts)
```

Comment:
I agree that we'll want something more sophisticated but there's also value in not making use of the next best complicated thing right away. The stats this PR outputs are very useful immediately (even if homegrown) for performance tuning and the rest can be done in a follow-up (I'd also like to see how exactly this is going to play with timeseries).